virtual_rainforest/core/logger.py

```diff
@@ -0,0 +1,31 @@
+"""The `core.logging` module.
+
+The `core.logging` module is used to setup the extend the standard logging setup to
+provide additional functionality relevant to the virtual rainforest model.
+
+At the moment the module simply sets up the logger so that other modules can access it.
+It is very likely to be further extended in future.
+"""
+
+import logging
+from typing import Type
+
+LOGGER = logging.getLogger("virtual_rainforest")
+
+
+def log_and_raise(msg: str, exception: Type[Exception], extra: dict = None) -> None:
```

Comment:
I think you can do this in one go using `LOGGER.error`. See example https://stackoverflow.com/a/37992988
That would save this whole function.

Reply:
Would it halt the execution of the program? Because I guess the advantage of defining this function is that it lets you log an error and halt the program with a single function call, rather than having to call `LOGGER.error` followed by `raise OSError` every time?

Reply:
You're right. It does not raise thee exception. So, good as it is!

Reply:
Phew! This function is lifted from another project that @jacobcook1995 and I work on 😄